go/coinstacks/cosmos/api/api.go

```diff
@@ -81,47 +81,47 @@ func New(httpClient *cosmos.HTTPClient, grpcClient *cosmos.GRPCClient, wsClient
 		logger.Panicf("%+v", err)
 	}
 
-	r.Use(api.Scheme)
-	r.Use(api.Logger)
+	router.Use(api.Scheme)
+	router.Use(api.Logger)
 
-	r.HandleFunc("/", a.Root).Methods("GET")
+	router.HandleFunc("/", apiRef.Root).Methods("GET")
 
-	r.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
+	router.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
 		api.HandleResponse(w, http.StatusOK, map[string]string{"status": "up", "coinstack": "cosmos", "connections": strconv.Itoa(manager.ConnectionCount())})
 	}).Methods("GET")
 
-	r.HandleFunc("/swagger", func(w http.ResponseWriter, r *http.Request) {
+	router.HandleFunc("/swagger", func(w http.ResponseWriter, r *http.Request) {
 		http.ServeFile(w, r, filepath.FromSlash(swaggerPath))
 	}).Methods("GET")
 
-	r.PathPrefix("/docs/").Handler(http.StripPrefix("/docs/", http.FileServer(http.Dir("./static/swaggerui"))))
+	router.PathPrefix("/docs/").Handler(http.StripPrefix("/docs/", http.FileServer(http.Dir("./static/swaggerui"))))
 
-	v1 := r.PathPrefix("/api/v1").Subrouter()
-	v1.HandleFunc("/info", a.Info).Methods("GET")
-	v1.HandleFunc("/send", a.SendTx).Methods("POST")
+	v1 := router.PathPrefix("/api/v1").Subrouter()
+	v1.HandleFunc("/info", apiRef.Info).Methods("GET")
+	v1.HandleFunc("/send", apiRef.SendTx).Methods("POST")
 
 	v1Account := v1.PathPrefix("/account").Subrouter()
 	v1Account.Use(cosmos.ValidatePubkey)
-	v1Account.HandleFunc("/{pubkey}", a.Account).Methods("GET")
-	v1Account.HandleFunc("/{pubkey}/txs", a.TxHistory).Methods("GET")
+	v1Account.HandleFunc("/{pubkey}", apiRef.Account).Methods("GET")
+	v1Account.HandleFunc("/{pubkey}/txs", apiRef.TxHistory).Methods("GET")
 
 	v1Transaction := v1.PathPrefix("/tx").Subrouter()
-	v1Transaction.HandleFunc("/{txid}", a.Tx).Methods("GET")
+	v1Transaction.HandleFunc("/{txid}", apiRef.Tx).Methods("GET")
 
 	v1Gas := v1.PathPrefix("/gas").Subrouter()
-	v1Gas.HandleFunc("/estimate", a.EstimateGas).Methods("POST")
+	v1Gas.HandleFunc("/estimate", apiRef.EstimateGas).Methods("POST")
 
 	v1Validators := v1.PathPrefix("/validators").Subrouter()
```

Comment:
we can either break this out into two subrouters for `/validators` and `/validator` to allow us to use the middleware on the `/validator` subrouter, or if we want to keep only a single subrouter, we will need to add the `cosmos.ValidateValidatorPubkey` to any handlers with `{pubkey}` path param